### PR TITLE
boards: infineon: add cy8ckit_062s2_ai support

### PR DESCRIPTION
**Pull request by SandraArrow**
This PR adds support for the cy8ckit_062s2_ai board which is based on CY8C624ABZI-S2D44 SoC. 
 
Infineon's customized OpenOCD version and pyOCD are supported as runner. 
 
The following samples were tested: 

- samples/hello_world
- samples/basic/blinky
- samples/basic/button

### CODE REVIEW

File: boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Enable console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# Enable UART driver
+CONFIG_SERIAL=y
+
+# Enable pin controller
+CONFIG_PINCTRL=y
+
```
Comment (nordicjm) [inline boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:23]: not allowed

File: boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2024 Arrow Electronics.
+# SPDX-License-Identifier: Apache-2.0
+
+# General configuration
+CONFIG_CORTEX_M_SYSTICK=y
+CONFIG_BUILD_OUTPUT_HEX=y
```
Comment (nordicjm) [inline boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:6]: @gmarull

File: boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+CONFIG_PINCTRL=y
+
+# Enable GPIO driver
+CONFIG_GPIO=y
+
+# Enable clock controller
+CONFIG_CLOCK_CONTROL=y
+
+# Main Stack Size
+CONFIG_MAIN_STACK_SIZE=2048
```
Comment (nordicjm) [inline boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:31]: why is this needed?

File: boards/infineon/cy8ckit_062s2_ai/doc/img/cy8ckit_062s2_ai.png
Comment (nordicjm) [inline boards/infineon/cy8ckit_062s2_ai/doc/img/cy8ckit_062s2_ai.png:1]: Convert to webp, then put through https://tinypng.com/

File: boards/infineon/cy8ckit_062s2_ai/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+- `CY8C624ABZI-S2D44 MCU SoC Website`_
+- `CY8C624ABZI-S2D44 MCU Datasheet`_
+- `CY8CKIT-062S2-AI Website`_
+- `CY8CKIT-062S2-AI User Guide`_
+- `CY8CKIT-062S2-AI Schematics`_
+
+Supported Features
+==================
+
+The board configuration supports the following hardware features:
```
Comment (nordicjm) [inline boards/infineon/cy8ckit_062s2_ai/doc/index.rst:31]: use proper name for building with board with \`\` around it, see https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-terminology for terminology

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig (hunk 2 above)
Comment (gmarull) [reply boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:6]: these do not seem board specific configs but soc level preferences. Systick may not even be needed if enabled in DT.

File: boards/infineon/cy8ckit_062s2_ai/doc/index.rst (hunk 4 above)
Comment (SandraArrow) [reply boards/infineon/cy8ckit_062s2_ai/doc/index.rst:31]: Sorry, I'm not sure if I understood you comment correctly. Can you give me an example what I should change where?

File: boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig (hunk 2 above)
Comment (SandraArrow) [reply boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:5]: I copied it from another PSoC board. I deleted `CONFIG_CORTEX_M_SYSTICK=y`, but `CONFIG_BUILD_OUTPUT_HEX=y` is required to flash an application with (Infineon's version of) openOCD.

Without this Kconfig option, you get the following error:

```shell
-- west flash: rebuilding
ninja: no work to do.
-- west flash: using runner openocd
FATAL ERROR: hex file location is unknown. Try enabling CONFIG_BUILD_OUTPUT_HEX.
```

Comment (gmarull) [reply boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:5]: > `CONFIG_BUILD_OUTPUT_HEX=y` is required to flash an application with (Infineon's version of) openOCD.

This means other boards are actually wrong, this soc cannot be easily flashed if there's no hex in the build folder. Add this option in the soc defconfig.

Comment (SandraArrow) [reply boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:5]: This seems to be a more generic issue in the PSoC SoC configurations. 
I checked the other supported PSoC boards. Almost all supported PSoC boards (cyw920829m2evk_02, cy8cproto_063_ble, cy8cproto_062_4343w) cannot be flashed with Infineon's openOCD version, if you comment out this Kconfig option in the board defconfig file. For cy8ckit_062s4 only pyOCD is supported as runner and this is able to flash bin files.
I am not sure if this issue should be fixed in this pull request since this Kconfig option is missing in all PSoC SoC configurations. I think, it makes more sense to fix it in a separate pull request and then also delete it in all board defconfig files. 
I have also done a code search and found this Kconfig option in around 50 board defconfig files. So, it seems that there are several SoC configurations that do not enable this option.

Comment (keithmwheeler) [reply boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:5]: CONFIG_OUTPUT_BUILD_HEX=y is in all Cypress and Infineon PSoC6 board defconfigs; this is a requirement with our version of openOCD -- and as you've noted with many other boards as well. Is the suggestion that this should move from the board defconfig to an SOC file?

Comment (SandraArrow) [reply boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:5]: Yes, this was the suggestion from gmarull to move this Kconfig option from the board defconfig to the SoC defconfig file. However, I think, it makes no sense to change it just for one PSoC SoC (family). If we change it, we should change it for all PSoC families and then delete it in all PSoC board configurations.
On the other side, you find this Kconfig option in a lot of board configurations. So, I am not sure if it needs be changed.
@gmarull What do you think about it?

Comment (gmarull) [reply boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig:5]: I think it's ok to ask to do this right for the board you're adding, other boards can be fixed later.

Comment (SandraArrow) [reply boards/infineon/cy8ckit_062s2_ai/cy8ckit_062s2_ai_defconfig]: I have now deleted this Kconfig option in the board defconfig file and added it to the  `soc/infineon/cat1a/psoc6_02/Kconfig.defconfig` file, as there is no separate defconfig file for each SoC. This Kconfig option is now set for all PSOC 6_02 based MCUs.

**Review by nordicjm — DISMISSED** [review]

File: boards/infineon/cy8ckit_062s2_ai/board.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2024 Arrow Electronics.
+# SPDX-License-Identifier: Apache-2.0
+
+board:
+  name: cy8ckit_062s2_ai
+  full_name: PSOC6 AI Evaluation Kit
```
Comment (kartben) [inline boards/infineon/cy8ckit_062s2_ai/board.yml:6]: Pointing this one out as it's important that trademarks are properly used/spelled, especially as this appears in the board listing and can be used to search for boards by name

```suggestion
  full_name: PSOC 6 AI Evaluation Kit
```

**Review by kartben — CHANGES_REQUESTED** [review]
One minor request, looks great otherwise, thanks!

**Review by npal-cy — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Comment by kartben** [comment]
@ifyall please take a look

**Review by ifyall — APPROVED** [review]

**Review by kartben — APPROVED** [review]
